modules/vaos/app/services/ccra/referral_service.rb

```diff
@@ -65,8 +65,28 @@ def get_referral(id, icn)
         # Note JSON.parse is only used for betamock responses
         data = response.body.is_a?(String) ? JSON.parse(response.body, symbolize_names: true) : response.body
 
-        ReferralDetail.new(data)
+        referral = ReferralDetail.new(data)
+        cache_referral_data(id, referral)
```

Comment:
This needs to change from `id` to `referral.referral_number`. EPS does not have the consult ID.